darts/models/forecasting_model.py

```diff
@@ -254,12 +259,20 @@ def historical_forecasts(self,
                 train_cov = covariates.drop_after(pred_time)
 
             if retrain:
-                if covariates and "covariates" in fit_signature.parameters:
-                    self.fit(series=train, covariates=train_cov)
-                elif covariates and "exog" in fit_signature.parameters:
-                    self.fit(series=train, exog=train_cov)
+                if self.__class__.__bases__[0].__name__ == 'TorchForecastingModel':
```

Comment:
@hrzn since `epoch` parameter to the `fit()` method is unique to the `TorchForecastingModels` I have founded my code based on base class name